addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -75,15 +96,216 @@ public void importFromUrl(final String url) throws IllegalArgumentException, IOE
+                    contentType = contentTypeMap.getOrDefault(language.toLowerCase(), "text/html");
+                }
+            }
+
+            bodyContent = body.getRaw();
+        } else if (mode.equals("urlencoded")) {
+            contentType = "application/x-www-form-urlencoded";
+
+            StringBuilder urlencodedBodySB = new StringBuilder();
+
+            for (KeyValueData data : body.getUrlencoded()) {
+                if (!data.isDisabled()) {
+                    if (urlencodedBodySB.length() > 0) {
+                        urlencodedBodySB.append("&");
+                    }
+                    try {
+                        urlencodedBodySB
+                                .append(
+                                        URLEncoder.encode(
+                                                data.getKey(), StandardCharsets.UTF_8.name()))
+                                .append("=")
+                                .append(
+                                        URLEncoder.encode(
+                                                data.getValue(), StandardCharsets.UTF_8.name()));
+                    } catch (UnsupportedEncodingException e) {
+                    }
+                }
+            }
+
+            bodyContent = urlencodedBodySB.toString();
+        } else if (mode.equals("formdata")) {
+            String boundary = "----" + System.currentTimeMillis();
+
+            contentType = "multipart/form-data; boundary=" + boundary;
+
+            StringBuilder formDataBodySB = new StringBuilder();
+            for (FormData formData : body.getFormData()) {
+                if (!formData.isDisabled()) {
+                    formDataBodySB.append(generateMultiPartBody(formData, boundary)).append(CRLF);
+                }
+            }
+
+            formDataBodySB.append("--").append(boundary).append("--").append(CRLF);
+
+            bodyContent = formDataBodySB.toString();
+        } else if (mode.equals("file")) {
+            String src = body.getFile().getSrc();
+
+            contentType = getFileContentType(src);
+
+            try {
+                bodyContent = FileUtils.readFileToString(new File(src), StandardCharsets.UTF_8);
+            } catch (IOException e1) {
+            }
+        } else if (mode.equals("graphql")) {
+            contentType = "application/json";
+
+            GraphQl graphQlBody = body.getGraphQl();
+            String query = graphQlBody.getQuery().replaceAll("\r\n", "\\\\r\\\\n");
+            String variables = graphQlBody.getVariables().replaceAll("\\s", "");
+
+            bodyContent = String.format("{\"query\":\"%s\", \"variables\":%s}", query, variables);
+        }
+
+        if (!isContentTypeAlreadySet(request.getHeader())) {
+            httpMessage.getRequestHeader().setHeader("Content-Type", contentType);
+        }
+
+        httpMessage.getRequestBody().setBody(bodyContent.toString());
+        httpMessage.getRequestHeader().setContentLength(bodyContent.length());
```

Comment:
Use `httpMessage.getRequestBody().length()` instead, number of bytes might be difference from the number of characters.